src/moonlink/src/storage/cache/object_storage/cache_config.rs

```diff
@@ -16,25 +19,13 @@ impl ObjectStorageCacheConfig {
     }
 
     /// Provide a default option for ease of testing.
+    /// It requires to take a testcase-unique temporary directory.
     #[cfg(test)]
-    pub fn default_for_test() -> Self {
+    pub fn default_for_test(temp_dir: &TempDir) -> Self {
         const DEFAULT_MAX_BYTES_FOR_TEST: u64 = 1 << 30; // 1GiB
-        const DEFAULT_CACHE_DIRECTORY: &str = "/tmp/moonlink_test_cache";
```

Comment:
`cargo test` executes all test cases in a parallel fashion, default cache recreation is subject to data race.